public/components/agents/vuls/inventory.tsx

```diff
@@ -51,9 +90,64 @@ export class Inventory extends Component {
     this._isMount = false;
   }
 
+  async fetchVisualizationVulnerabilitiesSummaryData(field, agentID) {
+    const results = await getAggregation(agentID, field, 4);
+    return Object.entries(results[field]).map(([key, value], index) => ({
+      label: key,
+      value,
+      color: this.colorsVisualizationVulnerabilitiesSummaryData[index],
+      onClick: () => this.onFiltersChange(this.buildFilterQuery(field, key))
+    })).sort((firstElement, secondElement) => secondElement.value - firstElement.value)
+  }
+
+  async fetchVisualizationVulnerabilitiesSeverityData(){
+    const { id } = this.props.agent;
+    const FIELD = 'severity';
+    this.setState({ isLoadingStats: true });
+    //Assign a value to the severity label to be able to sort it by level
+    const severityOrder = { Critical: 12, High: 9, Medium: 6, Low: 3 };
+
+    const vulnerabilityLastScan = await getLastScan(id);
+    const { severity } = await getAggregation(id, FIELD);
+
+    const severityStats = Object.keys(severityOrder).map(key => ({ 
+      titleColor: this.titleColors[key],
+      description: key,
+      title: severity[key] ? severity[key] : 0
+    }))
+    .sort((prev, next) => severityOrder[prev.description] > severityOrder[next.description] ? -1 : 1);
+
+    this.setState({
+      stats: severityStats,
+      isLoadingStats: false,
+      vulnerabilityLastScan
+    });
+    
+    return Object.entries(severity)
+      .map(([key, value]) => ({
+        label: key,
+        value,
+        color: this.titleColors[key],
+        onClick: () => this.onFiltersChange(this.buildFilterQuery(FIELD, key))
+      }
+      ))
+      .sort((prev, next) => severityOrder[prev.label] > severityOrder[next.label] ? -1 : 1);
```

Comment:
**nitpick**: Sorting depending on the label of severity level could not be required if we predefine the order. See https://github.com/wazuh/wazuh-kibana-app/pull/3893/files#r829768322.